src/components/core/PivotTable.tsx

```diff
@@ -313,19 +316,25 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
             if (this.isGrowToFitEnabled(prevProps) !== this.isGrowToFitEnabled()) {
                 this.growToFit(this.columnApi);
             }
-            const prevColumnWidths = this.getColumnWidths(prevProps);
-            const columnWidths = this.getColumnWidths(this.props);
-            if (!isEqual(prevColumnWidths, columnWidths) && !this.waitingForFirstExecution) {
-                if (this.columnApi) {
-                    this.resizedColumnsStore.updateColumnWidths(columnWidths, this.getExecutionResponse());
 
-                    syncSuppressSizeToFitOnColumns(this.resizedColumnsStore, this.columnApi);
-
-                    if (this.isGrowToFitEnabled()) {
-                        this.growToFit(this.columnApi); // calls resetColumnsWidthToDefault internally too
+            if (!isEqual(prevColumnWidths, columnWidths) && !this.columnWidthsChangeWaitingForExecution) {
+                if (this.columnApi) {
+                    if (this.shouldWaitForExecution(totalsPropsChanged, totalsStateChanged)) {
+                        this.columnWidthsChangeWaitingForExecution = true;
                     } else {
-                        const columns = this.columnApi.getAllColumns();
-                        this.resetColumnsWidthToDefault(this.columnApi, columns);
+                        this.resizedColumnsStore.updateColumnWidths(
```

Comment:
The contents of this else branch seems to me a rather deep, consider if it would be convenient to separate to a method.

Reply:
refactored